cmd/bacalhau/run_test.go

```diff
@@ -0,0 +1,199 @@
+			)
+			assert.NoError(suite.T(), err, "Error submitting job. Run - Number of Jobs: %s. Job number: %s", tc.numberOfJobs, i)
+
+			job, _, err := c.Get(ctx, strings.TrimSpace(out))
+			assert.NoError(suite.T(), err)
+			assert.NotNil(suite.T(), job, "Failed to get job with ID: %s", out)
+			// assert.Equal(suite.T(), tc.numberOfJobsOutput, strings.Count(out, "\n"))
+		}()
+	}
+}
+
+func (suite *RunSuite) TestRun_CreatedAt() {
+	tests := []struct {
+		numberOfJobs int
+	}{
+		{numberOfJobs: 1}, // Test for one
+		{numberOfJobs: 5}, // Test for five
+	}
+
+	for i, tc := range tests {
+		func() {
+			ctx := context.Background()
+			c, cm := publicapi.SetupTests(suite.T())
+			defer cm.Cleanup()
+
+			parsedBasedURI, _ := url.Parse(c.BaseURI)
+			host, port, _ := net.SplitHostPort(parsedBasedURI.Host)
+			_, out, err := ExecuteTestCobraCommand(suite.T(), suite.rootCmd, "run",
+				"--api-host", host,
+				"--api-port", port,
+				"ubuntu echo 'hello world'",
+			)
+			assert.NoError(suite.T(), err, "Error submitting job. Run - Number of Jobs: %s. Job number: %s", tc.numberOfJobs, i)
+
+			job, _, err := c.Get(ctx, strings.TrimSpace(out))
+			assert.NoError(suite.T(), err)
+			assert.NotNil(suite.T(), job, "Failed to get job with ID: %s", out)
+			assert.LessOrEqual(suite.T(), job.CreatedAt, time.Now(), "Created at time is not less than or equal to now.")
+
+			oldStartTime, _ := time.Parse(time.RFC3339, "2021-01-01T01:01:01+00:00")
+			assert.GreaterOrEqual(suite.T(), job.CreatedAt, oldStartTime, "Created at time is not greater or equal to 2022-01-01.")
+		}()
+
+	}
+}
+func (suite *RunSuite) TestRun_Annotations() {
+	// TODO: Assign to Aronchick - change tests to just reject if annotation is bad.
+	suite.T().Skip()
+
+	tests := []struct {
+		numberOfJobs int
+	}{
+		{numberOfJobs: 1}, // Test for one
+		// {numberOfJobs: 5}, // Test for five
+	}
+
+	annotationsToTest := []struct {
+		Name          string
+		Annotations   []string
+		CorrectLength int
+		BadCase       bool
+	}{
+		{Name: "1", Annotations: []string{""}, CorrectLength: 0, BadCase: false},               // Label flag, no value, but correctly quoted
+		{Name: "1.1", Annotations: []string{`""`}, CorrectLength: 1, BadCase: false},           // Label flag, no value, but correctly quoted
+		{Name: "2", Annotations: []string{"a"}, CorrectLength: 1, BadCase: false},              // Annotations, string
+		{Name: "3", Annotations: []string{"a", "1"}, CorrectLength: 2, BadCase: false},         // Annotations, string and int
+		{Name: "4", Annotations: []string{`''`, `" "`}, CorrectLength: 2, BadCase: false},      // Annotations, some edge case characters
+		{Name: "5", Annotations: []string{"🏳", "0", "🌈️"}, CorrectLength: 3, BadCase: false},   // Emojis
+		{Name: "6", Annotations: []string{"ايطاليا"}, CorrectLength: 1, BadCase: false},        // Right to left
+		{Name: "7", Annotations: []string{"‫test‫"}, CorrectLength: 1, BadCase: false},         // Control charactel
```

Comment:
this line of code is a bit scary looking

Reply:
<img width="912" alt="CleanShot 2022-06-24 at 19 41 28@2x" src="https://user-images.githubusercontent.com/51317/175614055-9a42036c-6347-4334-955e-34253e2f9d2c.png">